### DISCUSSION TITLE
Extend Twill's User Model

### DISCUSSION THREAD

**Discussion by zipavlin** [area17/twill#1002] [Ideas]
Is it possible to add relationship to Twill's User model?

I checked #130, but I would like to extend existing class - which gives me 'Class should not extend itself' error - basically I just like to add two methods, instead of overwriting whole model.

**Comment by ifox** [comment]
Hi @zipavlin!

That is currently not possible without overwriting the file through Composer configuration which is not a recommended approach. We are seeing more and more need for Twill internal models extensibility so I'm leaning towards extracting some of them into configuration properties. That way you should able to extend it in your own namespace and only add your 2 methods.
